app/ui/src/app/integration/integration_detail/integration_activity/integration-activity.component.ts

```diff
@@ -97,6 +99,17 @@ export class IntegrationActivityComponent implements OnInit {
     this.showPagination = (this.allActivities.length > this.paginationConfig.pageSize);
     this.paginationConfig.totalItems = this.allActivities.length;
     this.paginationConfig.pageNumber = 1;
+
+    this.openshiftConsoleURL = null;
+    if (this.allActivities.length > 0) {
+      const base = this.configService.getSettings('consoleUrl');
+      const pod = this.allActivities[0].pod;
+      const project = this.configService.getSettings('project');
```

Comment:
Same goes with project, would it make more sense to keep that stored in the integration object rather than baking it into the config.json?

Reply:
Integrations can be exported/imported into different projects. Also, from an architecturale POV an integration does not need know about the OpenShift details it is  running in. All integration runtimes also share the same OpenShift project, so an integration is not the proper place to store this info. 

Of course this information could (and imo should) come from the server, too (as, btw, all of config.json, not sure why there is more configuration than the connection to the backend stored in config.json as everything else is probably better managed on the server side. even branding).

But for now thats the simplest and easiest way without querying OpenShift itself or changing the way how configuration information reaches the frontend.

Reply:
Actually, instead of calculating the URL here in the frontend which requires exposing random bits of backend information, why don't we just have a 'logUrl' field in the integration object and just do this URL building at the backend?  This current solution feels super-brittle, if we merge it as-is we're gonna need to create issue(s) to go back in and fix it the right way.

Reply:
A syndesis install right now is fixed to a single project.  An integration can be exported between enviroments.  The pod id, is can vary over time within a single environment.  The base console URL is variable between environments.  We could do the computation for you with a new rest API, but that would have the same brittle qualities as doing it in the frontend.

Reply:
Suppose maybe the best option is a GET that does a 302 redirect to openshift console page, then the anchor link would be like `<a href="/v1/integrations/{id}/consoleLog">` then no need to expose any kind of openshift-y stuff to the frontend.

Reply:
that sounds valid to me, and I agree we should not put backend config to the client. @chirino any idea if this is hard to implement ? I guess its only about getting the pod id as the base URL and project can be injected in the templates as well to the server config.

@gashcrumb the only problem is that this URL should only appear _if_ the consoleUrl is configured. So we would need another endpoint to query this before rendering the link (or not).

Reply:
gashcrumb that might not work in the future.. the UI may need to add #anchor bits to the URL which I'm not sure work with a redirect.  To me the seem squarely a Syndesis UI to OCP Console UI concern.  Not sure why the backend APIs needs to be burdened with UI concerns.

Reply:
Yes, a document hash can be used as part of a 302 redirect, the hash isn't sent to the server.  Actually the openshift console uses this feature to pass along it's auth token to the java console to avoid transmitting the auth token.

I'm just giving options.  The current proposed code means you can potentially have a stale link at any point should the integration get restarted.  Really the link only has to work in the short time the user clicks it, so calculating it from the backend at the point of handling the GET request where I'm guessing the information is more easily obtained seems a win.

Reply:
@gashcrumb the possible staleness was also a concern to me, but we discussed the issue with a stale link at it is acceptable to the PM if this happens, right @kcbabo ? It can get stale for many reasons. Also the link might not be valid at all, so the backend would have to redirect to a proper error message (however would be nice if the link was not visible from the very beginning when this is the case or gets automatically updated when the situation changes, like we do for other integration states via webhooks).

As I mentioned in the original mail thread:

> This is only the case if the pod scales up, which only happens if the user manually scales the pods post-deployment, right?


> It's also the case when the pod dies or is restarted (e.g. because of healthchecks or because). Its also true when the use stops the pod (because it stop the integrations) or makes a new deployment for a new revision or change revision. The link can get stale anytime (because such things can happen in the background when the use is already on the page). This is highly dynamic, also when such a link can be enabled/disable. The only good technical solution would be to enable such a link all the time, but this is a link to the UI which doese a rest request to the backend for determing the currently running pod and the either redirect to the OpenShift console (but be aware about authnetication !) or show an error dialog if no such pod exists. A static link would not work for the above reasons.